labelme/widgets/canvas.py

```diff
@@ -476,15 +485,18 @@ def selectShapePoint(self, point, multiple_selection_mode):
         else:
             for shape in reversed(self.shapes):
                 if self.isVisible(shape) and shape.containsPoint(point):
-                    self.calculateOffsets(shape, point)
-                    self.setHiding()
-                    if multiple_selection_mode:
-                        if shape not in self.selectedShapes:
+                    if shape not in self.selectedShapes:
+                        self.setHiding()
+                        if multiple_selection_mode:
                             self.selectionChanged.emit(
                                 self.selectedShapes + [shape]
                             )
+                        else:
+                            self.selectionChanged.emit([shape])
+                        self.wasSelected = False
                     else:
-                        self.selectionChanged.emit([shape])
+                        self.wasSelected = True
+                    self.calculateOffsets(self.selectedShapes, point)
```

Comment:
This should be shape instead of selectedShapes?

Reply:
Yes, I missed that `self.calculateOffsets` will changed in another PR to enable operating with multiple selected shapes.